src/serializer/utils.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+/* @flow */
+
+import type { ObjectValue } from "../values/index.js";
+import type { Realm } from "../realm.js";
+
+import invariant from "../invariant.js";
+import { IsArray } from "../methods/index.js";
+
+/**
+ * Get index property list length by searching array properties list for the max index key value plus 1.
+ */
+export function getArrayIndexPropertyLength(realm: Realm, val: ObjectValue): number {
```

Comment:
The name is confusing. But I can't think of a really good one either. If you don't do the +1 in here, and return -1 if no entries, then the name `maxKnownArrayIndex` would make sense.